modules/src/ics03_connection/version.rs

```diff
@@ -70,22 +70,26 @@ pub fn get_compatible_versions() -> Vec<Version> {
 pub fn pick_version(
     supported_versions: Vec<Version>,
     counterparty_versions: Vec<Version>,
-) -> Option<Version> {
+) -> Result<Version, Error> {
```

Comment:
Unrelated to this PR, but I think `Version::features` would be better modelled as a [HashSet](https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html) instead of a `Vec`. This also makes intersections easier.

Reply:
I think the caller of this method, namely `ConnectionReader::pick_version`, receives the two versions as vectors, cf. https://github.com/cosmos/ibc/tree/master/spec/core/ics-003-connection-semantics#versioning. It would be an improvement indeed to transform these vectors (the caller `ConnectionReader::pick_version` could do that) into hashsets, and pass them as sets to this `pick_version` method here.

Since this is unrelated to this PR, let's keep it in mind for later (potentially open an issue to track it, since it improves code hygiene).